scripts/update-versions.js

```diff
@@ -1,62 +1,56 @@
 const fs = require('fs');
-const glob = require('glob');
+const { globSync } = require('glob');
 
-// Get command line arguments
 // Get command line arguments
 const fileNamePatterns = process.argv.slice(2, -2);
 const targetKey = process.argv[process.argv.length - 2];
 const replacementValue = process.argv[process.argv.length - 1];
 for (const fileNamePattern of fileNamePatterns) {
 	// Find matching JSON files in current directory and all subdirectories
-	glob(
-		`**/${fileNamePattern}`,
-		{ nodir: true, recursive: true, ignore: '**/node_modules/**' },
-		(matchError, files) => {
-			if (matchError) {
-				console.error(matchError);
+	const files = globSync(`**/${fileNamePattern}`, {
+		nodir: true,
+		recursive: true,
+		ignore: '**/node_modules/**'
+	});
+
+	// Loop through each file
+	for (const file of files) {
+		// Read JSON file
+		fs.readFile(file, 'utf8', (readError, data) => {
+			if (readError) {
+				console.error(readError);
 				return;
 			}
 
-			// Loop through each file
-			for (const file of files) {
-				// Read JSON file
-				fs.readFile(file, 'utf8', (readError, data) => {
-					if (readError) {
-						console.error(readError);
-						return;
-					}
-
-					// Parse JSON object
-					const json = JSON.parse(data);
-
-					// Modify target key
-					if (targetKey === 'runtime' && json[targetKey].version) {
-						json[targetKey].version = replacementValue;
-					}
-
-					if (targetKey === 'devDependencies' && json[targetKey]['openfin-adapter']) {
-						json[targetKey]['openfin-adapter'] = replacementValue;
-					}
-
-					if (targetKey === 'devDependencies' && json[targetKey]['@openfin/core']) {
-						json[targetKey]['@openfin/core'] = replacementValue;
-					}
-
-					if (targetKey === 'version' && json['name'] === 'openfin-container-starter') {
-						json[targetKey] = replacementValue;
-					}
-
-					// Write updated JSON object back to file
-					fs.writeFile(file, JSON.stringify(json, null, 2), (writeError) => {
-						if (writeError) {
-							console.error(writeError);
-							return;
-						}
-
-						console.log(`Successfully updated ${targetKey} in ${file}`);
-					});
-				});
+			// Parse JSON object
+			const json = JSON.parse(data);
+
+			// Modify target key
+			if (targetKey === 'runtime' && json[targetKey].version) {
+				json[targetKey].version = replacementValue;
 			}
-		}
-	);
+
+			if (targetKey === 'devDependencies' && json[targetKey]['openfin-adapter']) {
```

Comment:
if the replacementValue is set once as the runtime for the manifest shouldn't it be shorter for the openfin-adapter and core types? It might be it isn't showing in the diff.

Reply:
@johnman do you mean 


```suggestion
			if (targetKey === 'devDependencies' && (json[targetKey]['openfin-adapter'] || json[targetKey]['@openfin/core']) ) {
```

Reply:
- name: Update Runtime Version
        run: node scripts/update-versions.js *.manifest.*.json manifest.*.json *.fin.json runtime ${{ inputs.manifestVersion }}

      - name: Update Package & Adapter Version
        run: node scripts/update-versions.js package.json devDependencies ${{ inputs.packageVersion }}

      - name: Update main repo version
        run: node scripts/update-versions.js package.json version ${{ inputs.packageVersion }}

the second one is for the main package json